src/confluent_kafka/schema_registry/json_schema.py

```diff
@@ -313,7 +376,15 @@ def __call__(self, data, ctx):
             obj_dict = json.loads(payload.read())
 
             try:
-                validate(instance=obj_dict, schema=self._parsed_schema)
+                if self._are_references_provided:
+                    registered_schema = self._registry.get_schema(schema_id)
```

Comment:
Check why this is required?

Reply:
Thanks for catching this. It's not needed as the schema has been already passed in the constructor. Added a field to store the passed schema and using that here.